### PR TITLE
fix(cpp/db):  how to win friends and infleunce enemies missing dialogue

### LINKED ISSUE #20110: [DK Starter Chain Quest] How to Win Friends and Influence Enemies. Missing dialogues during the torture.
### Current Behaviour

The quest can be completed, but dialogues are not displayed during torture process.
In addition i'm not sure that other npcs should attack tortured npc when it become friendly to the player.

Evidence: [Youtube](https://youtu.be/0--rYRaqmaM)

### Expected Blizzlike Behaviour

The Npc and the player should talk during the torture.
How it should work: [Youtube](https://www.youtube.com/watch?v=B21NbuUlinw) from minute 33:40

### Source

_No response_

### Steps to reproduce the problem

1. Start the quest (ID: 12720)
2. Start fighting Scarlet npcs with Keleseth's Persuader equipped

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/784e80668d5b57640a4c7a4d6e2c7b898e4f804d

### Operating system

Windows 11

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by ConorB1745**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

Scarlet Crusader (id: 28940), Scarlet Marksman (id: 28610), Scarlet Preacher (id: 28939) and Scarlet Commander (id: 28936) were missing dialog in the database and script npc_crusade_persuaded always succeeded when the spell on Keleseth's Persuader (item id: 39371 and spell id: 52781) procced.

The success chance has been reduced from 100% to 30% based on retail wow testing. The player characer now always says a line when the spell procs on the weapon. If it's successful the roleplay starts for the scarlet, if it fails the scarlet says a line refusing to reveal the information.

Additional lines of dialogue were added from Wowpedia: https://wowpedia.fandom.com/wiki/Scarlet_Crusader_(mob)

NB: While updating the script some code within the class was updated to remove magic numbers.

## Issues Addressed:

- Closes #20110

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

1) Create a new Death Knight character of any race/faction.
2) Complete all the quests up to the point of  receiving the quest "How to Win Friends and Infleunce Enemies". This is the quest under test.
3) Torture scarlets until your hearts content.

Helpful commands:
.quest complete 12593 
.quest complete 12619
.quest complete 12842
.quest complete 12848
.quest complete 12636
.quest complete 12641
.quest complete 12657
.quest complete 12850
.quest complete 12670
.quest complete 12678
.quest complete 12680
.quest complete 12679
.quest complete 12733
.quest complete 12687 -> Shadow realm do this if it teleports after hand in.
.quest complete 12697
.quest complete 12698
.quest complete 12700
.quest complete 12701
.quest complete 12706
.quest complete 12714
.quest complete 12715
.quest complete 12719

Run around pick up the quests, type the above commands, and then hand in to get to the point quicker. Some quests are pickup and hand in by default but are still included above.

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+VALUES(77372, 0, 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019),
+(77373, 0, 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019),
+(77374, 0, 'Look out! There''s a bee loose and it''s on a stinging rampage!', 'Look out! There''s a bee loose and it''s on a stinging rampage!', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019),
+(77375, 0, 'That one tickled...', 'That one tickled...', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019),
+(77376, 0, 'I''m going to send you back to the Lich King in pieces!', 'I''m going to send you back to the Lich King in pieces!', 0, 0, 0, 0, 0, 0, 0, 0, 1, 18019);
+
+INSERT INTO acore_world.creature_text
```
Comment (Winfidonarleyan) [inline data/sql/updates/pending_db_world/rev_1728059831124992100.sql:9]:
```suggestion
INSERT INTO creature_text
```

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+(CreatureID, GroupID, ID, `Text`, `Type`, `Language`, Probability, Emote, Duration, Sound, BroadcastTextId, TextRange, comment)
+VALUES(28939, 1, 6, 'You... You''ll get nothing... And like it...', 12, 0, 100.0, 0, 0, 0, 77372, 0, 'crusader SAY_CURSADER7'),
+(28939, 1, 7, 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 12, 0, 100.0, 0, 0, 0, 77373, 0, 'crusader SAY_CURSADER8'),
+(28939, 1, 8, 'Look out! There''s a bee loose and it''s on a stinging rampage!', 12, 0, 100.0, 0, 0, 0, 77374, 0, 'crusader SAY_CURSADER9'),
+(28939, 1, 9, 'That one tickled...', 12, 0, 100.0, 0, 0, 0, 77375, 0, 'crusader SAY_CURSADER10'),
+(28939, 1, 10, 'I''m going to send you back to the Lich King in pieces!', 12, 0, 100.0, 0, 0, 0, 6, 0, 'crusader SAY_CURSADER11');
+
+INSERT INTO acore_world.creature_text
```
Comment (Winfidonarleyan) [inline data/sql/updates/pending_db_world/rev_1728059831124992100.sql:17]:
```suggestion
INSERT INTO creature_text
```

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+(CreatureID, GroupID, ID, `Text`, `Type`, `Language`, Probability, Emote, Duration, Sound, BroadcastTextId, TextRange, comment)
+VALUES(28940, 1, 6, 'You... You''ll get nothing... And like it...', 12, 0, 100.0, 0, 0, 0, 77372, 0, 'crusader SAY_CURSADER7'),
+(28940, 1, 7, 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 12, 0, 100.0, 0, 0, 0, 77373, 0, 'crusader SAY_CURSADER8'),
+(28940, 1, 8, 'Look out! There''s a bee loose and it''s on a stinging rampage!', 12, 0, 100.0, 0, 0, 0, 77374, 0, 'crusader SAY_CURSADER9'),
+(28940, 1, 9, 'That one tickled...', 12, 0, 100.0, 0, 0, 0, 77375, 0, 'crusader SAY_CURSADER10'),
+(28940, 1, 10, 'I''m going to send you back to the Lich King in pieces!', 12, 0, 100.0, 0, 0, 0, 77376, 0, 'crusader SAY_CURSADER11');
+
+INSERT INTO acore_world.creature_text
```
Comment (Winfidonarleyan) [inline data/sql/updates/pending_db_world/rev_1728059831124992100.sql:25]:
```suggestion
INSERT INTO creature_text
```

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+(CreatureID, GroupID, ID, `Text`, `Type`, `Language`, Probability, Emote, Duration, Sound, BroadcastTextId, TextRange, comment)
+VALUES(28610, 1, 6, 'You... You''ll get nothing... And like it...', 12, 0, 100.0, 0, 0, 0, 77372, 0, 'crusader SAY_CURSADER7'),
+(28610, 1, 7, 'Do you think this to be pain? To be suffering? HAH! Destroy this mortal shell so that I may ascend to the heavens!', 12, 0, 100.0, 0, 0, 0, 77373, 0, 'crusader SAY_CURSADER8'),
+(28610, 1, 8, 'Look out! There''s a bee loose and it''s on a stinging rampage!', 12, 0, 100.0, 0, 0, 0, 77374, 0, 'crusader SAY_CURSADER9'),
+(28610, 1, 9, 'That one tickled...', 12, 0, 100.0, 0, 0, 0, 77375, 0, 'crusader SAY_CURSADER10'),
+(28610, 1, 10, 'I''m going to send you back to the Lich King in pieces!', 12, 0, 100.0, 0, 0, 0, 77376, 0, 'crusader SAY_CURSADER11');
+
+INSERT INTO acore_world.creature_text
```
Comment (Winfidonarleyan) [inline data/sql/updates/pending_db_world/rev_1728059831124992100.sql:33]:
```suggestion
INSERT INTO creature_text
```

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp [hunk 5]
```diff
@@ -81,20 +90,23 @@ class npc_crusade_persuaded : public CreatureScript
             {
                 if (Player* player = caster->ToPlayer())
                 {
-                    if (player->GetQuestStatus(QUEST_HOW_TO_WIN_FRIENDS) == QUEST_STATUS_INCOMPLETE)
+                    playerGUID = player->GetGUID();
+                    speechTimer = SPEECH_TIMER_DEFAULT;
+                    speechCounter = SPEECH_COUNTER_START_SUCCESS_ROLEPLAY;
+                    uint32 persuadeRoll = std::rand() % 10 + 1;
```
Comment (Winfidonarleyan) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp:96]: using urand instead

Comment (ConorB1745) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp]: Changed to use urand

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql (hunk 4 above)
Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: Removed acore_world from INSERT INTO

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql (hunk 3 above)
Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: Removed acore_world from INSERT INTO

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql (hunk 2 above)
Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: Removed acore_world from INSERT INTO

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql (hunk 1 above)
Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: Removed acore_world from INSERT INTO

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 6]
```diff
@@ -0,0 +1,62 @@
+INSERT INTO acore_world.broadcast_text
```
[the remark this answers is not in the document]
Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: Removed acore_world from INSERT INTO

**Comment by Nyeriah** [comment]
Broadcast_text changes cannot be accepted I am sorry.

**Comment by Nyeriah** [comment]
Broadcast_text changes cannot be accepted I am sorry.

**Comment by ConorB1745** [comment]
> Broadcast_text changes cannot be accepted I am sorry.

Yeah I understand I wasn't sure how to tackle the fail on that check because it meant the removal of the new/missing text. If I remove the broadcast text insert and new creature text insert but keep the script changes and Scarlet Commander (CreatureId 28936) inserts (without new broadcast requiring ones) will it be valid? @Nyeriah

**Comment by Rorschach91** [comment]
I have tested this PR and It seems that It works.
It can be reviewed.

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 7]
```diff
@@ -0,0 +1,25 @@
+INSERT INTO creature_text
+(CreatureID, GroupID, ID, `Text`, `Type`, `Language`, Probability, Emote, Duration, Sound, BroadcastTextId, TextRange, comment)
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1728059831124992100.sql:2]: Before insert add always a delete, moreover add the backtick in all columns and in the table name

Comment (ConorB1745) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: Hey, I've implemented the changes mentioned here. Is it okay to group the deletes into one query like that or should it have been an individual delete for each?

Comment (Helias) [reply data/sql/updates/pending_db_world/rev_1728059831124992100.sql]: I think that what you did is ok and don't have other better options

File: data/sql/updates/pending_db_world/rev_1728059831124992100.sql [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+(28936, 8, 1, 'Tell me what you know about "Crimson Dawn" or the beatings will continue!', 12, 0, 100.0, 0, 0, 0, 29134, 0, 'player SAY_PERSUADE2'),
+(28936, 8, 2, 'I''m through being courteous with your kind, human! What is the "Crimson Dawn?"', 12, 0, 100.0, 0, 0, 0, 29135, 0, 'player SAY_PERSUADE3'),
+(28936, 8, 3, 'Is your life worth so little? Just tell me what I need to know about "Crimson Dawn" and I''ll end your suffering quickly.', 12, 0, 100.0, 0, 0, 0, 29139, 0, 'player SAY_PERSUADE4'),
+(28936, 8, 4, 'I can keep this up for a very long time, Scarlet dog! Tell me about the "Crimson Dawn!"', 12, 0, 100.0, 0, 0, 0, 29137, 0, 'player SAY_PERSUADE'),
+(28936, 8, 5, 'What is the "Crimson Dawn?"', 12, 0, 100.0, 0, 0, 0, 29133, 0, 'player SAY_PERSUADE6'),
+(28936, 8, 6, '"Crimson Dawn!" What is it! Speak!', 12, 0, 100.0, 0, 0, 0, 29136, 0, 'player SAY_PERSUADE7');
+
+UPDATE `creature_template` SET `ScriptName` = 'npc_crusade_persuaded' WHERE (`entry` = 28940);
+
+UPDATE `item_template` SET `spellppmRate_1` = 8 WHERE (`entry` = 39371);
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1728059831124992100.sql:27]:
```suggestion
UPDATE `item_template` SET `spellppmRate_1` = 8 WHERE (`entry` = 39371);

```

**Review by Helias — APPROVED** [review]

**Pull request merged**